registry/storage/driver/middleware/cloudfront/s3filter.go

```diff
@@ -0,0 +1,267 @@
+package middleware
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	dcontext "github.com/docker/distribution/context"
+	"github.com/hashicorp/go-version"
+)
+
+const (
+	// ipRangesURL is the URL to get definition of AWS IPs
+	ipRangesURL = "https://ip-ranges.amazonaws.com/ip-ranges.json"
+	// updateFrequency tells how frequently AWS IPs need to be updated
+	updateFrequency = time.Hour * 12
+)
+
+// newAWSIPs returns a New aWSIP object.
+// If awsRegion is `nil`, it accepts any region. Otherwise, it only allow the regions specified
+func newAWSIPs(host string, updateFrequency time.Duration, awsRegion []string) *awsIPs {
+	ips := &awsIPs{
+		host:            host,
+		updateFrequency: updateFrequency,
+		awsRegion:       awsRegion,
+		updaterStopChan: make(chan bool),
+	}
+	ips.tryUpdate()
```

Comment:
Something should be done with this error, what is the significance of this initially failing? Should it return nil or error out the initailization